### ISSUE TITLE
Inconsistent behavior when session expires (refresh token dead)

### ISSUE THREAD

**Issue by dcristolovean** [aws-amplify/amplify-swift#3943]
### Describe the bug

I'm transitioning from AWSMobileClient to Amplify and it's a huge endeavor. I'm a bit stuck with handling session expiration cases and it might be a bug (or intended behavior, but then see the questions).
My app is configured to use auth user AND non-auth users.

So, let's assume in my Amplify.Hub.listen (...) I receive the sessionExpired event. 
This leaves me in a totally inconsistent state, where, I would like to revert to a non-auth user and continue using the app.   Or display something, doesn't matter.

But when I get the sessionExpired event, nothing works anymore: fetchAuthSession is dead, getIdentityId is dead, API calls are dead.
And the reason for that is that the fetchAuthSession returns isSignedIn = true and still has a user stalled somewhere.

The only way I could transition back to a non auth user is to call Amplify.Auth.signOut() and actually get rid of my bad user.  After that, everything works again, I am a non-auth user.

This approach has one big problem:  the user plays in the app and suddenly he gets the system dialog to sign in.  We all know the issue with that text in the system dialog.  It's really bad to just show this dialog out of the blue. And they might press CANCEL.

So... how can I get rid of the logged in user with the expired session ? Silently, not showing anything to the user.  

### Steps To Reproduce

```swift
Kinda hard to provide some steps, it's a more generic theoretical question
```

### Expected behavior

1. Either clean up the user when sessionExpires automatically and make sure we have an non-auth user working.
2. Provide another signOut() way without web. I use SSO and I need the cookies.  This user is actually local, the session is dead anyway, so not really sure what a webpage is supposed to do with it.  

### Amplify Framework Version

2.45.2

### Amplify Categories

Auth

### Dependency manager

Swift PM

### Swift version

latest

### CLI version

no CLI

### Xcode version

latest

### Relevant log output

```shell
<details>
<summary>Log Messages</summary>


INSERT LOG MESSAGES HERE
```
</details>
```


### Is this a regression?

Yes

### Regression additional context

I think it's like this since forever.

### Platforms

iOS

### OS Version

iOS 18

### Device

any

### Specific to simulators

_No response_

### Additional context

_No response_

**Comment by dcristolovean** [comment]
I discovered also an interesting behavior.  So the session is dead and I call signOut().  I press CANCEL on the system dialog but the user is logged out and session can be fixed now, with fetch.

Which leads me to believe that signOut() has 2 parts:  one on API side, web stuff, it works if there's a session valid, otherwise ....  and one local part that cleans up the local user.   The thing is that if the 1st part fails (or i press cancel), you still clean up local stuff and everything will work from now on.

At least there's that, but if I'm right, then you really should have a signOut() method that actually detects the session is dead so there's no point for any web stuff and only do the local part.

**Comment by dcristolovean** [comment]
I will test at some point, but not right now, I got the code into a very stable version and can't risk it at the moment.
BUT:  as long as Amplify does the logout silently locally when session is expired, without opening a useless webpage and prompting the user with the wrong apple text,  we're golden! :)

**Comment by dcristolovean** [comment]
yes, I know the story with the text.  But if it's silent and we no longer see it, all is perfect.

**Comment by harsh62** [comment]
The issue was fixed and released.
